library/src/main/java/com/datatorrent/lib/projection/ProjectionOperator.java

```diff
@@ -0,0 +1,303 @@
+ * selectFields and dropFields are optional and either of them shall be specified
+ * When both are not specified, all fields shall be projected to downstream operator
+ *
+ * <b>Input Port</b> takes POJOs as an input
+ *
+ * <b>Output Ports</b>
+ * - projected port emits POJOs with projected fields from input POJOs
+ * - remainder port, if connected, emits POJOs with remainder fields from input POJOs
+ * - error port emits input POJOs as is upon error situations
+ * 
+ * <b>Examples</b>
+ * For {a, b, c} type of input tuples
+ *  - when selectFields = "" and dropFields = "", projected port shall emit {a, b, c}
+ *  - when selectFields = "b", projected port shall emit {b} and remainder port shall emit {a, c}
+ *  - when dropFields = "b", projected port shall emit {a, c} and remainder port shall emit {b}
+ * 
+ */
+
+public class ProjectionOperator extends BaseOperator implements Operator.ActivationListener<Context>
+{
+  protected String selectFields;
+  protected String dropFields;
+  protected String condition;
+
+  static class TypeInfo
+  {
+    String name;
+    Class type;
+    PojoUtils.Setter setter;
+    PojoUtils.Getter getter;
+
+    public TypeInfo(String name, Class<?> type)
+    {
+      this.name = name;
+      this.type = type;
+    }
+
+    public String toString()
+    {
+      String s = new String("'name': " + name + " 'type': " + type);
+      return s;
+    }
+  }
+
+  protected transient List<TypeInfo> projectedFields = new ArrayList<>();
+  protected transient List<TypeInfo> remainderFields = new ArrayList<>();
+
+  @AutoMetric
+  protected long projectedTuples;
+
+  @AutoMetric
+  protected long remainderTuples;
+
+  @AutoMetric
+  protected long errorTuples;
+
+  protected Class<?> inClazz = null;
+  protected Class<?> projectedClazz = null;
+  protected Class<?> remainderClazz = null;
+
+  @InputPortFieldAnnotation(schemaRequired = true)
+  public transient DefaultInputPort<Object> in = new DefaultInputPort<Object>()
+  {
+    public void setup(PortContext context)
+    {
+      inClazz = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+
+    @Override
+    public void process(Object t)
+    {
+      handleProjection(t);
+    }
+  };
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> projected = new DefaultOutputPort<Object>()
+  {
+    public void setup(PortContext context)
+    {
+      projectedClazz = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+  };
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> remainder = new DefaultOutputPort<Object>()
+  {
+    public void setup(PortContext context)
+    {
+      remainderClazz = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+  };
+
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> error = new DefaultOutputPort<Object>()
+  {
+    public void setup(PortContext context)
```

Comment:
Not sure if this is required to be done.
inClass is already set by this point or will be set in the future for sure.